src/vmm/src/builder.rs

```diff
@@ -730,13 +707,39 @@ fn attach_legacy_devices_aarch64(
         .map_err(VmmError::RegisterMMIODevice)
 }
 
-fn create_vcpus(vm: &Vm, vcpu_count: u8, exit_evt: &EventFd) -> Result<Vec<Vcpu>, VmmError> {
+// For x86_64 we need to create the interrupt controller before calling `KVM_CREATE_VCPUS`
+// while on aarch64 we need to do it the other way around.
+#[cfg(target_arch = "x86_64")]
+fn create_vcpus(vm: &mut Vm, vcpu_count: u8, exit_evt: &EventFd) -> Result<Vec<Vcpu>, VmmError> {
+    setup_interrupt_controller(vm)?;
+
+    let mut vcpus = Vec::with_capacity(vcpu_count as usize);
+    for cpu_idx in 0..vcpu_count {
+        let exit_evt = exit_evt.try_clone().map_err(VmmError::EventFd)?;
+        let vcpu = Vcpu::new(cpu_idx, vm, exit_evt).map_err(VmmError::VcpuCreate)?;
+        vcpus.push(vcpu);
+    }
+    // Make stdout non blocking.
+    set_stdout_nonblocking();
```

Comment:
This seems to belong to the serial device init, so can be moved back to pio device manager init code.

Reply:
The aarch64::attach_legacy_devices function also relies on this functionality. Moving it into the PIO device manager would make implementation significantly more challenging.

Reply:
How about move this into the `setup_serial_device`? I see that we always call `set_stdout_nonblocking` before attaching the serial.